File: .NET/Microsoft.Recognizers.Text.Sequence/Extractors/BasePhoneNumberExtractorConfiguration.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System.Collections.Generic;
using System.Text.RegularExpressions;
using Microsoft.Recognizers.Definitions;
using Microsoft.Recognizers.Definitions.Utilities;

namespace Microsoft.Recognizers.Text.Sequence
{
    public class BasePhoneNumberExtractorConfiguration : PhoneNumberConfiguration
    {
        public BasePhoneNumberExtractorConfiguration(SequenceOptions options)
            : base(options)
        {
            WordBoundariesRegex = BasePhoneNumbers.WordBoundariesRegex;
            NonWordBoundariesRegex = BasePhoneNumbers.NonWordBoundariesRegex;
            EndWordBoundariesRegex = BasePhoneNumbers.EndWordBoundariesRegex;
            ColonPrefixCheckRegex = new Regex(BasePhoneNumbers.ColonPrefixCheckRegex);
            ColonMarkers = (List<char>)BasePhoneNumbers.ColonMarkers;
            ForbiddenPrefixMarkers = (List<char>)BasePhoneNumbers.ForbiddenPrefixMarkers;
            ForbiddenSuffixMarkers = (List<char>)BasePhoneNumbers.ForbiddenSuffixMarkers;
            AmbiguityFiltersDict = AmbiguityFiltersDict = DefinitionLoader.LoadAmbiguityFilters(BasePhoneNumbers.AmbiguityFiltersDict);
        }
    }
}

File: Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/extractors/BasePhoneNumberExtractorConfiguration.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package com.microsoft.recognizers.text.sequence.extractors;

import com.microsoft.recognizers.text.sequence.SequenceOptions;
import com.microsoft.recognizers.text.sequence.config.PhoneNumberConfiguration;
import com.microsoft.recognizers.text.sequence.resources.BasePhoneNumbers;

import java.util.regex.Pattern;

public class BasePhoneNumberExtractorConfiguration extends PhoneNumberConfiguration {
    public BasePhoneNumberExtractorConfiguration(SequenceOptions options) {
        super(options);
        setWordBoundariesRegex(BasePhoneNumbers.WordBoundariesRegex);
        setNonWordBoundariesRegex(BasePhoneNumbers.NonWordBoundariesRegex);
        setEndWordBoundariesRegex(BasePhoneNumbers.EndWordBoundariesRegex);
        setColonPrefixCheckRegex(Pattern.compile(BasePhoneNumbers.ColonPrefixCheckRegex));
        setColonMarkers(BasePhoneNumbers.ColonMarkers);
        setForbiddenPrefixMarkers(BasePhoneNumbers.ForbiddenPrefixMarkers);
        setForbiddenSuffixMarkers(BasePhoneNumbers.ForbiddenSuffixMarkers);
    }
}
